Distributions/UnitTests/Interpolation/wlTestInterpolation.F90

```diff
@@ -0,0 +1,215 @@
+PROGRAM wlTestInterpolation
+
+  USE wlKindModule, ONLY: dp
+  USE wlInterpolationModule, ONLY: &
+    LinearInterp_Array_Point, &
+    GetIndexAndDelta
+
+  USE, INTRINSIC :: iso_fortran_env, only : stdin=>input_unit, &
+                                            stdout=>output_unit, &
+                                            stderr=>error_unit
+
+  IMPLICIT NONE
+
+  !--------- parameters for creating grid to test interpolation ------------
+  INTEGER, PARAMETER     :: nx_lo_res = 15 !create two different base arrays to check 
+  INTEGER, PARAMETER     :: nx_hi_res = 30 !for convergence of the interpolation routines
+  
+  !--------- extents of the test array
+  REAL(dp), PARAMETER    :: x1_min = 0.0d0
+  REAL(dp), PARAMETER    :: x1_max = 1.0d0
+  REAL(dp), PARAMETER    :: x2_min = -1.0d0
+  REAL(dp), PARAMETER    :: x2_max = 2.0d0
+  REAL(dp), PARAMETER    :: x3_min = 2.0d0
+  REAL(dp), PARAMETER    :: x3_max = 3.0d0
+  REAL(dp), PARAMETER    :: x4_min = -2.5d0
+  REAL(dp), PARAMETER    :: x4_max = 2.7d0
+  REAL(dp), PARAMETER    :: x5_min = 0.0d0
+  REAL(dp), PARAMETER    :: x5_max = 10.0d0
+
+  !--------- the actual points to interpolate
+  REAL(dp), PARAMETER    :: x1 = 0.1d0
+  REAL(dp), PARAMETER    :: x2 = -0.34d0
+  REAL(dp), PARAMETER    :: x3 = 2.47d0
+  REAL(dp), PARAMETER    :: x4 = -0.793d0
+  REAL(dp), PARAMETER    :: x5 = 9.375d0
+
+  REAL(dp), DIMENSION(nx_lo_res,nx_lo_res,nx_lo_res,nx_lo_res,nx_lo_res) :: linear_data
+
+  REAL(dp), DIMENSION(nx_lo_res,nx_lo_res,nx_lo_res,nx_lo_res,nx_lo_res) :: non_linear_data_lo_res
+  REAL(dp), DIMENSION(nx_hi_res,nx_hi_res,nx_hi_res,nx_hi_res,nx_hi_res) :: non_linear_data_hi_res
+
+  REAL(dp), DIMENSION(nx_lo_res) :: axis1_lo_res
+  REAL(dp), DIMENSION(nx_lo_res) :: axis2_lo_res
+  REAL(dp), DIMENSION(nx_lo_res) :: axis3_lo_res
+  REAL(dp), DIMENSION(nx_lo_res) :: axis4_lo_res
+  REAL(dp), DIMENSION(nx_lo_res) :: axis5_lo_res
+
+  REAL(dp), DIMENSION(nx_hi_res) :: axis1_hi_res
+  REAL(dp), DIMENSION(nx_hi_res) :: axis2_hi_res
+  REAL(dp), DIMENSION(nx_hi_res) :: axis3_hi_res
+  REAL(dp), DIMENSION(nx_hi_res) :: axis4_hi_res
+  REAL(dp), DIMENSION(nx_hi_res) :: axis5_hi_res
+
+  INTEGER :: ii, jj, kk, ll, mm
+
+  REAL(dp) :: dx1_lo, dx1_hi
+  REAL(dp) :: dx2_lo, dx2_hi
+  REAL(dp) :: dx3_lo, dx3_hi
+  REAL(dp) :: dx4_lo, dx4_hi
+  REAL(dp) :: dx5_lo, dx5_hi
+
+  REAL(dp) :: interp_result_linear
+  REAL(dp) :: interp_result_non_linear_lo
+  REAL(dp) :: interp_result_non_linear_hi
+
+  INTEGER  :: ix1, ix2, ix3, ix4, ix5
+  REAL(dp) :: dx1, dx2, dx3, dx4, dx5
+
+  REAL(dp) :: linear_value
+  REAL(dp) :: non_linear_value
+
+  dx1_lo = (x1_max-x1_min) / DBLE(nx_lo_res-1)
```

Comment:
`REAL(nx_lo_res-1,dp)` is technically "more correct", since `dx1_lo` is type `dp`, but they are equivalent for all intents and purposes. Only mention this because I think the XL Fortran compiler will scream at this.